From 5a49c973a05fd387cbd012c3db918572179bd9c7 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sat, 25 Apr 2020 13:56:38 +0200
Subject: [PATCH 1/4] backport nv_ioremap_nocache changes from 440.64

---
 common/inc/nv-linux.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/common/inc/nv-linux.h b/common/inc/nv-linux.h
index 3bce0c0..20459bd 100644
--- a/common/inc/nv-linux.h
+++ b/common/inc/nv-linux.h
@@ -575,10 +575,7 @@ static inline void *nv_ioremap(NvU64 phys, NvU64 size)
 
 static inline void *nv_ioremap_nocache(NvU64 phys, NvU64 size)
 {
-    void *ptr = ioremap_nocache(phys, size);
-    if (ptr)
-        NV_MEMDBG_ADD(ptr, size);
-    return ptr;
+    return nv_ioremap(phys, size);
 }
 
 static inline void *nv_ioremap_cache(NvU64 phys, NvU64 size)
-- 
2.20.1

