Author: Andreas Beckmann <anbe@debian.org>
Description: run conftest.sh from within Kbuild

--- a/nvidia-modules-common.mk
+++ b/nvidia-modules-common.mk
@@ -254,7 +254,7 @@ KBUILD_PARAMS += ARCH=$(ARCH)
 # End of Makefile section
 endif
 
-#ifneq ($(KERNELRELEASE),)
+ifneq ($(KERNELRELEASE),)
 # Begin of Kbuild section
 
 #
@@ -285,8 +285,11 @@ build-sanity-checks: conftest-verbose
 	 fi; \
 	done
 
+# For any object files that depend on conftest, declare the dependency here.
+$(addprefix $(obj)/,$(MODULE_GLUE_OBJS)): build-sanity-checks $(CONFTEST_HEADERS)
+
 # End of Kbuild section
-#endif
+endif
 
 ifeq ($(KERNELRELEASE),)
 # Begin of Makefile section
@@ -295,7 +298,7 @@ ifeq ($(KERNELRELEASE),)
 # Call into KBUILD to build the NVIDIA kernel module(s) defined in $(obj-m)
 #
 
-module: build-sanity-checks $(CONFTEST_HEADERS)
+module:
 	@echo "NVIDIA: calling KBUILD..."
 	@$(MAKE) $(KBUILD_PARAMS) modules
 	@echo "NVIDIA: left KBUILD."
