Author: Andreas Beckmann <anbe@debian.org>
Description: run conftest.sh from within Kbuild

--- a/Makefile
+++ b/Makefile
@@ -234,6 +234,7 @@ $(obj)/patches.h: $(src)/conftest.sh con
 
 RESMAN_GLUE_TARGETS = $(addprefix $(obj)/,$(RESMAN_GLUE_OBJS))
 
+$(RESMAN_GLUE_TARGETS): build-sanity-checks
 $(RESMAN_GLUE_TARGETS): $(obj)/conftest.h $(obj)/patches.h
 
 $(obj)/nv.o: $(obj)/$(VERSION_HEADER)
@@ -297,7 +298,7 @@ ifeq ($(KERNELRELEASE),)
 # the "package-install" target below.
 #
 
-module: build-sanity-checks
+module:
 	@echo "NVIDIA: calling KBUILD..."
 	@$(MAKE) $(KBUILD_PARAMS) modules
 	@echo "NVIDIA: left KBUILD."
