Author: Andreas Beckmann <anbe@debian.org>
Description: dump the generated conftest headers

Index: kernel/Kbuild
===================================================================
--- kernel.orig/Kbuild
+++ kernel/Kbuild
@@ -121,6 +121,16 @@ NV_CONFTEST_HEADERS += $(obj)/conftest/h
 NV_CONFTEST_HEADERS += $(NV_CONFTEST_COMPILE_TEST_HEADERS)
 
 
+.PHONY: conftest-verbose
+conftest-verbose:
+	@echo 'NV_CONFTEST_CMD=$(NV_CONFTEST_CMD)'
+	@echo 'NV_CONFTEST_CFLAGS=$(NV_CONFTEST_CFLAGS)'
+	@echo 'KBUILD_CFLAGS=$(KBUILD_CFLAGS)'
+	@echo 'LINUXINCLUDE=$(LINUXINCLUDE)'
+	@echo 'LDFLAGS=$(LDFLAGS)'
+	@echo 'KBUILD_LDFLAGS=$(KBUILD_LDFLAGS)'
+	@echo 'ARCH=$(ARCH)'
+
 #
 # Generate a header file for a single conftest compile test. Each compile test
 # header depends on conftest.sh, as well as the generated conftest/headers.h
@@ -145,6 +155,8 @@ define NV_GENERATE_COMPILE_TEST_HEADER
 	@mkdir -p $(obj)/conftest
 	@# concatenate /dev/null to prevent cat from hanging when $$^ is empty
 	@cat $$^ /dev/null > $$@
+	@echo " CONFTEST $$(notdir $$@):"
+	@cat $$@
 endef
 
 #
@@ -164,9 +176,11 @@ $(eval $(call NV_GENERATE_COMPILE_TEST_H
 $(eval $(call NV_GENERATE_COMPILE_TEST_HEADER,symbols,$(NV_CONFTEST_SYMBOL_COMPILE_TESTS)))
 $(eval $(call NV_GENERATE_COMPILE_TEST_HEADER,types,$(NV_CONFTEST_TYPE_COMPILE_TESTS)))
 
-$(obj)/conftest/patches.h: $(NV_CONFTEST_SCRIPT)
+$(obj)/conftest/patches.h: $(NV_CONFTEST_SCRIPT) conftest-verbose
 	@mkdir -p $(obj)/conftest
 	@$(NV_CONFTEST_CMD) patch_check > $@
+	@echo " CONFTEST $(notdir $@):"
+	@cat $@
 
 
 # Each of these headers is checked for presence with a test #include; a
@@ -246,9 +260,11 @@ NV_HEADER_PRESENCE_PART = $(addprefix $(
 
 # Define a rule to check the header $(1).
 define NV_HEADER_PRESENCE_CHECK
- $$(call NV_HEADER_PRESENCE_PART,$(1)): $$(NV_CONFTEST_SCRIPT)
+ $$(call NV_HEADER_PRESENCE_PART,$(1)): $$(NV_CONFTEST_SCRIPT) conftest-verbose
 	@mkdir -p $$(dir $$@)
 	@$$(NV_CONFTEST_CMD) test_kernel_header '$$(NV_CONFTEST_CFLAGS)' '$(1)' > $$@
+	@echo " CONFTEST $(notdir $@):"
+	@cat $@
 endef
 
 # Evaluate the rule above for each header in the list.
