Author: Andreas Beckmann <anbe@debian.org>
Description: make silent commands in NVIDIA's Makefile.kbuild verbose

--- a/nvidia-modules-common.mk
+++ b/nvidia-modules-common.mk
@@ -139,10 +139,10 @@ MOD_SIGN_CMD ?= $(KERNEL_SOURCES)/script
 #
 
 $(obj)/$(CORE_OBJS):
-	@cp $(src)/$(CORE_OBJS) $(obj)/$(CORE_OBJS)
+	$(NVQ)cp $(src)/$(CORE_OBJS) $(obj)/$(CORE_OBJS)
 
 $(obj)/$(VERSION_HEADER):
-	@echo \#define NV_COMPILER \"`$(CC) -v 2>&1 | tail -n 1`\" > $@
+	$(NVQ)echo \#define NV_COMPILER \"`$(CC) -v 2>&1 | tail -n 1`\" > $@
 
 CONFTEST_COMPILE_TEST_HEADERS := $(obj)/conftest/macros.h \
 $(obj)/conftest/functions.h $(obj)/conftest/symbols.h $(obj)/conftest/types.h \
@@ -224,10 +224,10 @@ KBUILD_PARAMS += ARCH=$(ARCH)
 #
 
 suser-sanity-check:
-	@if ! $(CONFTEST) suser_sanity_check; then exit 1; fi
+	$(NVQ)if ! $(CONFTEST) suser_sanity_check; then exit 1; fi
 
 rmmod-sanity-check:
-	@if ! $(CONFTEST) rmmod_sanity_check; then exit 1; fi
+	$(NVQ)if ! $(CONFTEST) rmmod_sanity_check; then exit 1; fi
 
 BUILD_SANITY_CHECKS = \
 	cc_version_check \
@@ -240,7 +240,7 @@ BUILD_SANITY_CHECKS = \
 build-sanity-checks:
 	@echo 'CONFTEST=$(CONFTEST)'
 	@echo 'CONFTEST_CFLAGS=$(CONFTEST_CFLAGS)'
-	@for SANITY_CHECK in $(BUILD_SANITY_CHECKS); do \
+	$(NVQ)for SANITY_CHECK in $(BUILD_SANITY_CHECKS); do \
 	 if ! $(CONFTEST) $$SANITY_CHECK full_output; then \
 	 exit 1; \
 	 fi; \
@@ -254,7 +254,7 @@ module: build-sanity-checks $(CONFTEST_H
 	@echo "NVIDIA: calling KBUILD..."
 	$(MAKE) $(KBUILD_PARAMS) modules
 	@echo "NVIDIA: left KBUILD."
-	for module in $(MODULE_OBJECTS); do \
+	$(NVQ)for module in $(MODULE_OBJECTS); do \
 	if ! [ -f $$module ]; then \
 	 echo "$$module failed to build!"; \
 	 exit 1; \
@@ -269,7 +269,7 @@ module: build-sanity-checks $(CONFTEST_H
 #
 
 module-sign: module
-	@for module in $(MODULE_OBJECTS); do \
+	$(NVQ)for module in $(MODULE_OBJECTS); do \
 	 if [ -z "$(MODSECKEY)" ] || [ -z "$(MODPUBKEY)" ]; then \
 	 echo "ERROR: MODSECKEY and MODPUBKEY must be set to sign modules!"; \
 	 exit 1; \
@@ -302,7 +302,7 @@ module-sign: module
 #
 
 module-install: suser-sanity-check module
-	@mkdir -p $(MODULE_ROOT)/video; \
+	$(NVQ)mkdir -p $(MODULE_ROOT)/video; \
 	install -m 0664 -o root -g root $(MODULE_OBJECTS) $(MODULE_ROOT)/video; \
 	PATH="$(PATH):/bin:/sbin" depmod -a
 
@@ -312,7 +312,7 @@ module-install: suser-sanity-check modul
 #
 
 package-install: module-install rmmod-sanity-check
-	@PATH="$(PATH):/bin:/sbin" modprobe $(MODULE_NAME) && \
+	$(NVQ)PATH="$(PATH):/bin:/sbin" modprobe $(MODULE_NAME) && \
 	echo "$(MODULE_OBJECTS) installed successfully.";
 
 #
@@ -335,17 +335,17 @@ MODULE_COMMON_SCRIPT := $(if $(wildcard
 #
 
 $(KERNEL_GLUE_NAME): module
-	@$(LD) $(EXTRA_LDFLAGS) $(MODULE_COMMON_SCRIPT) -r -o $(KERNEL_GLUE_NAME) $(KERNEL_GLUE_OBJS)
+	$(NVQ)$(LD) $(EXTRA_LDFLAGS) $(MODULE_COMMON_SCRIPT) -r -o $(KERNEL_GLUE_NAME) $(KERNEL_GLUE_OBJS)
 
 #
 # Support hack, KBUILD isn't prepared to clean up after external modules.
 #
 
 clean:
-	@$(RM) -f $(MODULE_GLUE_OBJS) $(KERNEL_GLUE_OBJS)
-	@$(RM) -f build-in.o nv-linux*.o *.d .*.cmd .*.flags
-	@$(RM) -f $(MODULE_NAME)*.o $(MODULE_NAME)*.ko*
-	@$(RM) -f $(MODULE_NAME)*.mod* $(VERSION_HEADER) *~
-	@$(RM) -f conftest*.c
-	@$(RM) -rf conftest
-	@$(RM) -rf Module*.symvers .tmp_versions modules.order
+	$(NVQ)$(RM) -f $(MODULE_GLUE_OBJS) $(KERNEL_GLUE_OBJS)
+	$(NVQ)$(RM) -f build-in.o nv-linux*.o *.d .*.cmd .*.flags
+	$(NVQ)$(RM) -f $(MODULE_NAME)*.o $(MODULE_NAME)*.ko*
+	$(NVQ)$(RM) -f $(MODULE_NAME)*.mod* $(VERSION_HEADER) *~
+	$(NVQ)$(RM) -f conftest*.c
+	$(NVQ)$(RM) -rf conftest
+	$(NVQ)$(RM) -rf Module*.symvers .tmp_versions modules.order
