From: Antonio Ospite <ospite@studenti.unina.it>
Description: Fix build with 2.6.38 kernels
Bug-Debian: http://bugs.debian.org/613798

Index: nvidia-260.19.21/nv-linux.h
===================================================================
--- nvidia-260.19.21.orig/nv-linux.h	2010-11-21 04:04:51.000000000 +0100
+++ nvidia-260.19.21/nv-linux.h	2011-02-16 19:44:47.000000000 +0100
@@ -1029,8 +1029,13 @@
 #endif
 
 #if defined(NV_ACQUIRE_CONSOLE_SEM_PRESENT)
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 37)
+#define NV_ACQUIRE_CONSOLE_SEM() console_lock()
+#define NV_RELEASE_CONSOLE_SEM() console_unlock()
+#else
 #define NV_ACQUIRE_CONSOLE_SEM() acquire_console_sem()
 #define NV_RELEASE_CONSOLE_SEM() release_console_sem()
+#endif
 #else
 #define NV_ACQUIRE_CONSOLE_SEM()
 #define NV_RELEASE_CONSOLE_SEM()
