Author: Andreas Beckmann <anbe@debian.org>
Description: separate Makefile/Kbuild specific parts within Makefile

--- a/nvidia-modules-common.mk
+++ b/nvidia-modules-common.mk
@@ -20,9 +20,16 @@
 module module-sign module-install package-install clean print-module-filename \
 conftest-compile-test
 
+ifeq ($(KERNELRELEASE),)
+# Begin of Makefile section
+
+modules: module
 all: install
 install: package-install
 
+# End of Makefile section
+endif
+
 #
 # The precompiled kernel module build process requires a separation of the
 # closed source and open source object files.
@@ -49,6 +56,9 @@ EXTRA_CFLAGS += -Wall -MD $(DEFINES) $(I
 src ?= .
 obj ?= $(src)
 
+ifeq ($(KERNELRELEASE),)
+# Begin of Makefile section
+
 #
 # Determine location of the Linux kernel source tree. Allow users to override
 # the default (i.e. automatically determined) kernel source location with the
@@ -82,6 +92,11 @@ else
  endif
 endif
 
+KBUILD_PARAMS	+= KERNEL_SOURCES=$(KERNEL_SOURCES) KERNEL_OUTPUT=$(KERNEL_OUTPUT)
+
+# End of Makefile section
+endif
+
 CC ?= cc
 HOST_CC ?= $(CC)
 LD ?= ld
@@ -112,6 +127,9 @@ TOPDIR ?= $(KERNEL_SOURCES)
 
 MODULE_OBJECT := $(MODULE_NAME).ko
 
+ifneq ($(KERNELRELEASE),)
+# Begin of Kbuild section
+
 #
 # NVIDIA specific CFLAGS and #define's.
 #
@@ -163,6 +181,12 @@ $(MODULE_GLUE_TARGETS): $(CONFTEST_HEADE
 
 $(obj)/nv.o: $(obj)/$(VERSION_HEADER)
 
+# End of Kbuild section
+endif
+
+ifeq ($(KERNELRELEASE),)
+# Begin of Makefile section
+
 #
 # KBUILD build parameters.
 #
@@ -179,6 +203,9 @@ KBUILD_PARAMS += KBUILD_VERBOSE=$(NV_VER
 KBUILD_PARAMS += -C $(KERNEL_SOURCES) M=$(PWD)
 KBUILD_PARAMS += ARCH=$(ARCH)
 
+# End of Makefile section
+endif
+
 #
 # NVIDIA sanity checks.
 #
@@ -189,6 +216,9 @@ suser-sanity-check:
 rmmod-sanity-check:
 	@if ! $(CONFTEST) rmmod_sanity_check; then exit 1; fi
 
+#ifneq ($(KERNELRELEASE),)
+# Begin of Kbuild section
+
 BUILD_SANITY_CHECKS = \
 	cc_version_check \
 	rivafb_sanity_check \
@@ -204,6 +234,12 @@ build-sanity-checks:
 	 fi; \
 	done
 
+# End of Kbuild section
+#endif
+
+ifeq ($(KERNELRELEASE),)
+# Begin of Makefile section
+
 #
 # BUILD_MODULE_RULE generates rules for building individual module targets.
 #
@@ -328,3 +364,6 @@ clean:
 
 print-module-filename:
 	@echo $(MODULE_OBJECT)
+
+# End of Makefile section
+endif
